### ISSUE TITLE
The question about mismatch of LER loss between paper and code

### ISSUE THREAD

**Issue by 1506775595hzy** [Cliffia123/LegoGCD#3]
Hi! Thanks for the inspiring work of GCD, but I have a quesion about the mismatch of LER loss between paper and code.

In the paper, LER loss is a  entropy loss with **a negative sign** (maximize the entropy) to "_encourage the predictions of above selected known samples close to a uniform distribution_", but I am confused about the code:
```
pseudo_label = torch.softmax((student_out/0.05), dim=-1)
logits_u = student_out/0.05
log_pred = F.log_softmax(logits_u, dim=-1)
nll_loss = torch.sum(-pseudo_label*log_pred, dim=1)*mask_old  # I think it is just a entropy loss
...(ect. )
loss += nll_loss.mean() # where should be loss -= nll_loss.mean()?
```
where the sign of the loss is mismatch with the fomular in the paper (missing a negatve sign), means that it reverse the original intention.
Does that means it mismatches with the paper? 

I would appreciate it if you could answer my question. Thank you!🙂

**Comment by Cliffia123** [comment]
Hi @1506775595hzy, thank you for your interest in our work and for pointing out this issue 😀.  The code is correct.  Actually, the negative sign is already included in the definition of the entropy **_H_**.  We will remove the negative sign in Eq. (10) and change the minus sign before $\Delta_j$ to a plus sign.

Thank you once again for highlighting this point.  We will update the formula in our paper and upload the revised version to arXiv shortly.

**Comment by 1506775595hzy** [comment]
Thank you for the reply! But I think you misunderstand my concern... 🤔
As mentioned above, **LER loss is an entropy loss with a negative sign (maximize the entropy)** to "encourage the predictions of the above selected known samples close to an **uniform distribution**", which means the performance is gained from maximizing the entropy (like the class mean entropy regulariser in SimGCD).

But you say that "the negative sign is already included in the definition of the entropy H", which means that LER tends to minimize the entropy loss and **dose not encourage a uniform distribution**, conflicting with the intention mentioned in the paper. 

This makes me very confused... I am sincerely hope you can answer my question.🙂

**Comment by Cliffia123** [comment]
@1506775595hzy Sorry for the late reply. We have repeatedly checked the logic of this paper and found that you are right. Now the original sentence in motivation is changed to:

Then,  we employ the information entropy function to encourage the predictions of above selected known samples close to a more  certain distribution, thereby increasing the confidence of samples from potential known classes.

However, it is worth noting that the modification of this sentence does not conflict with the conclusion in motivation
``Consequently, this LER prevents known samples from being misclassified as novel classes and therefore preserves the knowledge related to known categories during  learning novel categories.''

We will update the latest changes to the arxiv version. Thank you again for your interest in our work. ☺️

**Comment by 1506775595hzy** [comment]
Thanks for the clear response! >_< 
It made me better understand your pretty good work! 😘
